doc/webhook.md

```diff
@@ -0,0 +1,47 @@
+## Admission Webhook
+
+[Admission webhooks](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/) are HTTP callbacks that receive admission requests and do something with them.
+There are  two webhooks [ValidatingAdmissionWebhook](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#validatingadmissionwebhook) and 
+[MutatingAdmissionWebhook](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#mutatingadmissionwebhook) which are basically 
+doing the same thing except MutatingAdmissionWebhook can modify the requests. In our case, we use MutatingAdmissionWebhook.
```

Comment:
It may help to explain why we use a MutatingAdmissionWebhook instead of a validating one, since all we check here is if the upgrade is valid or not??

Reply:
Got it. Thanks.